events/UserAllocatedEvent.h

```diff
@@ -0,0 +1,460 @@
+    *
+    *  @param queue    Event queue to dispatch on. Will replace earlier bound EventQueue.
+    *
+    */
+    void call_on(EventQueue *queue)
+    {
+        MBED_ASSERT(!_post_ref);
+        MBED_UNUSED bool status = post_on(queue);
+        MBED_ASSERT(status);
+    }
+
+    /** Posts an event onto the underlying event queue
+    *
+    *  The event is posted to the event queue passed as argument
+    *  and is executed in the context of the event queue's dispatch loop.
+    *
+    *  This call cannot fail due queue memory exhaustion
+    *  because it doesn't allocate any memory
+    *
+    *  @return     False if the event was already posted
+    *              true otherwise
+    *
+    */
+    bool try_call()
+    {
+        return post();
+    }
+
+    /** Posts an event onto the event queue passed as argument,
+    *
+    *  The event is posted to the underlying queue and is executed in the
+    *  context of the event queue's dispatch loop.
+    *
+    *  This call cannot fail due queue memory exhaustion
+    *  because it doesn't allocate any memory
+    *
+    *  @param queue    Event queue to dispatch on. Will replace earlier bound EventQueue.
+    *  @return         False if the event was already posted
+    *                  true otherwise
+    *
+    */
+    bool try_call_on(EventQueue *queue)
+    {
+        return post(queue);
+    }
+
+    /** Posts an event onto the underlying event queue, returning void
+     *
+     *  The event is posted to the underlying queue and is executed in the
+     *  context of the event queue's dispatch loop.
+     *
+     *  This call cannot fail due queue memory exhaustion
+     *  because it doesn't allocate any memory
+     *
+     *  The post function is IRQ safe and can act as a mechanism for moving
+     *  events out of IRQ contexts.
+     *
+     */
+    void operator()()
+    {
+        return call();
+    }
+
+    /** Configure the delay of an event
+     *
+     *  @param delay    Millisecond delay before dispatching the event
+     */
+    void delay(int delay)
+    {
+        equeue_event_delay(&_e + 1, delay);
+    }
+
+    /** Configure the period of an event
+     *
+     *  @param period   Millisecond period for repeatedly dispatching an event
+     */
+    void period(int period)
+    {
+        equeue_event_period(&_e + 1, period);
+    }
+
+    /** Cancels posted event
+     *
+     *  Attempts to cancel posted event. It is safe to call
+     *  cancel after an event has already been dispatched.
+     *
+     *  The cancel function is IRQ safe.
+     *
+     *  If called while the event queue's dispatch loop is active, the cancel
+     *  function does not guarantee that the event will not execute after it
+     *  returns, as the event may have already begun executing.
+     *
+     *  @return     true if event was successfully cancelled
+     */
+    bool cancel() const
+    {
+        return equeue_cancel_user_allocated(_equeue, (struct equeue_event *)&_e);
```

Comment:
Why the cast?

Oh, const issue? `cancel()` shouldn't be a const method, should it?

Reply:
I think you are right, 
BTW `Event` has the same

Reply:
`Event`'s cancel is reasonably const, because it's not user-allocated, so the only change is inside the event queue.  (Indeed, shouldn't its post etc be const too...?)

Reply:
`const` removed